bson/_cbsonmodule.c

```diff
@@ -81,6 +81,98 @@ struct module_state {
 #define DATETIME_MS 3
 #define DATETIME_AUTO 4
 
+/* Converts integer to its string representation in decimal notation. */
+extern int cbson_long_long_to_str(long long num, char* str, size_t size) {
+    // Buffer should fit 64-bit signed integer
+    if (size < 21) {
+        PyErr_Format(
+            PyExc_RuntimeError,
+            "Buffer too small to hold long long: %d < 21", size);
+        return -1;
+    }
+    int index = 0;
+    int sign = 1;
+    // Convert to unsigned to handle -LLONG_MIN overflow
+    unsigned long long absNum;
+    // Handle the case of 0
+    if (num == 0) {
+        str[index++] = '0';
+        str[index] = '\0';
+        return 0;
+    }
+    // Handle negative numbers
+    if (num < 0) {
+        sign = -1;
+        absNum = 0ULL - (unsigned long long)num;
+    } else {
+        absNum = (unsigned long long)num;
+    }
+    // Convert the number to string
+    unsigned long long digit;
+    while (absNum > 0) {
+        digit = absNum % 10ULL;
+        str[index++] = (char)digit + '0';  // Convert digit to character
+        absNum /= 10;
+    }
+    // Add minus sign if negative
+    if (sign == -1) {
+        str[index++] = '-';
+    }
+    str[index] = '\0';  // Null terminator
+    // Reverse the string
+    int start = 0;
+    int end = index - 1;
+    while (start < end) {
+        char temp = str[start];
+        str[start++] = str[end];
+        str[end--] = temp;
+    }
+    return 0;
+}
+
+static PyObject* _test_long_long_to_str(PyObject* self, PyObject* args) {
+    // Test extreme values
+    Py_ssize_t maxNum = PY_SSIZE_T_MAX;
+    Py_ssize_t minNum = PY_SSIZE_T_MIN;
+    char str_1[BUF_SIZE];
+    char str_2[BUF_SIZE];
+    int res = LL2STR(str_1, (long long)minNum);
+    if (res == -1) {
+        return NULL;
+    }
+    INT2STRING(str_2, (long long)minNum);
+    if (strcmp(str_1, str_2) != 0) {
+        PyErr_Format(
+            PyExc_RuntimeError,
+            "LL2STR != INT2STRING: %s != %s", str_1, str_2);
+        return NULL;
+    }
+    LL2STR(str_1, (long long)maxNum);
+    INT2STRING(str_2, (long long)maxNum);
+    if (strcmp(str_1, str_2) != 0) {
+        PyErr_Format(
+            PyExc_RuntimeError,
+            "LL2STR != INT2STRING: %s != %s", str_1, str_2);
+        return NULL;
+    }
+
+    // Test common values
+    for (Py_ssize_t num = 0; num < 10000; num++) {
```

Comment:
`num` needs to be declared at the top of the function:
```
 [2023/06/05 17:06:32.104] gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Ibson -I/opt/_internal/cpython-3.7.16/include/python3.7m -c bson/_cbsonmodule.c -o build/temp.linux-x86_64-cpython-37/bson/_cbsonmodule.o
 [2023/06/05 17:06:32.150] + rm -rf build
 [2023/06/05 17:06:32.150] + /opt/python/cp37-cp37m/bin/python setup.py bdist_wheel
 [2023/06/05 17:06:32.150] bson/_cbsonmodule.c: In function ‘_test_long_long_to_str’:
 [2023/06/05 17:06:32.150] bson/_cbsonmodule.c:160:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
 [2023/06/05 17:06:32.150]      for (Py_ssize_t num = 0; num < 10000; num++) {
 [2023/06/05 17:06:32.150]      ^
 [2023/06/05 17:06:32.150] bson/_cbsonmodule.c:160:5: note: use option -std=c99 or -std=gnu99 to compile your code
```

Reply:
Done.